src/xenia/kernel/util/object_table.h

```diff
@@ -65,6 +76,18 @@ class ObjectTable {
     return results;
   }
 
+  template <typename T>
+  std::vector<object_ref<T>> GetObjectsByType() {
+    std::vector<object_ref<T>> results;
+    GetObjectsByType(
+        T::kType,
+        reinterpret_cast<std::vector<object_ref<XObject>>*>(&results));
+    return results;
+  }
+
+  std::vector<object_ref<XObject>> GetAllObjects();
+  void PurgeAllObjects(); // Purges the object table of all guest objects
```

Comment:
Linter's gonna yell at me for a lot of stuff :P

I'll do a commit when I'm done with these changes that lints everything properly.